addons/l10n_uy/models/account_move.py

```diff
@@ -48,3 +48,16 @@ def _get_l10n_latam_documents_domain(self):
                 allowed_codes = set(codes).intersection(set(matching_subtype_codes[0]))
                 domain += [("code", "in", tuple(allowed_codes))]
         return domain
+
+    def _compute_l10n_latam_document_type(self):
+        """
+        The following rules apply for determining document types based on the partner's identification:
+        RUT/RUC (Uruguay): Automatically select e-factura.
+        CI (Cedula de Identidad): Automatically select e-ticket, with the following considerations:
+        """
+        super()._compute_l10n_latam_document_type()
+        for rec in self:
+            if rec.partner_id.l10n_latam_identification_type_id == self.env.ref('l10n_uy.it_ci'):
+                rec.l10n_latam_document_type_id = self.env.ref('l10n_uy.dc_e_ticket').id
+            if rec.partner_id.l10n_latam_identification_type_id.name == self.env.ref('l10n_uy.it_rut'):
+                rec.l10n_latam_document_type_id = self.env.ref('l10n_uy.dc_e_inv').id
```

Comment:
Please add unit test to ensure that create a invoice with a contact with RUC will suggest e-factura. and create a invoice with a partner with CI will automatically select the e-ticket document.